api/clients/v2/verification/cert_verifier.go

```diff
@@ -107,6 +94,50 @@ func (cv *CertVerifier) CheckDACert(
 	return nil
 }
 
+// EstimateGasCheckDACert uses eth_estimateGas to estimate the gas requirements for a CheckDACert call.
+func (cv *CertVerifier) EstimateGasCheckDACert(
+	ctx context.Context,
+	cert coretypes.EigenDACert,
+) (uint64, error) {
+	// Normalize cert to V3
+	certV3 := NormalizeCertV3(cert)
+
+	certVerifierAddress, err := cv.addressProvider.GetCertVerifierAddress(
+		ctx,
+		certV3.ReferenceBlockNumber(),
+	)
+	if err != nil {
+		return 0, fmt.Errorf("get cert verifier address: %w", err)
+	}
+
+	certBytes, err := certV3.Serialize(coretypes.CertSerializationABI)
+	if err != nil {
+		return 0, fmt.Errorf("serialize cert: %w", err)
+	}
+
+	// Pack the checkDACert method call data
+	abi, err := certVerifierBinding.ContractEigenDACertVerifierMetaData.GetAbi()
+	if err != nil {
+		return 0, fmt.Errorf("get contract ABI: %w", err)
+	}
+
+	callData, err := abi.Pack("checkDACert", certBytes)
+	if err != nil {
+		return 0, fmt.Errorf("pack checkDACert call data: %w", err)
+	}
+
+	// Estimate gas using eth_estimateGas
+	gasEstimate, err := cv.ethClient.EstimateGas(ctx, ethereum.CallMsg{
+		To:   &certVerifierAddress,
+		Data: callData,
+	})
+	if err != nil {
+		return 0, fmt.Errorf("estimate gas for checkDACert: %w", err)
+	}
```

Comment:
oh just remembered. add a log.Error with the CallMsg arguments. We unfortunately don't have this in most of the codebase but we're trying to add this logging everywhere we do an rpc call. So that when they fail we can debug by going to tenderly or whatever